nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -568,4 +585,15 @@ public Task fetchTaskById(String documentId) {
         return (Task) getResponse(documentId);
     }
 
+    private void refuseIfDateFound(Object obj) {
+        if (obj instanceof Calendar || obj instanceof Date) {
+            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
+        }
+        if (obj instanceof Map) {
```

Comment:
Yes, if you want to be complete you should take into account lists and arrays (even though that's a rare case).